src/main/java/com/digitalpebble/storm/crawler/persistence/Status.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.digitalpebble.storm.crawler.persistence;
+
+public enum Status {
+    DISCOVERED, FETCHED, FETCH_ERROR, REDIR, ERROR;
```

Comment:
Will rename REDIR to REDIRECTION. As for ERROR vs FETCH_ERROR : the distinction is required so that the users can implement a different logic if they want to. Typically a FETCH ERROR could be due to a temporary problem with the network or the target server whereas ERROR would be used for any other case e.g. parser fails on a given document. With the former you want to reschedule the URL and give it another try later (up to a configurable number of tries), whereas with the latter you'd probably want to give it up straight away. That type of error could happen within the Fetcher as well and is not only for the parser.
Differentiating between the 2 gives users the option of implementing whichever logic the see fit.
Passing the name of the component would be an overkill IMHO + does not really address the fact that the Fetcher could generate both types of status.